components/Tabs/src/TabBarView/MDCTabBarView.m

```diff
@@ -443,73 +418,91 @@ - (void)layoutSubviews {
   }
 }
 
-- (void)willMoveToSuperview:(UIView *)newSuperview {
-  [super willMoveToSuperview:newSuperview];
-  self.initialScrollDone = NO;
+- (BOOL)isJustifiedLayoutStyle {
+  CGRect availableBounds = self.bounds;
+  if (@available(iOS 11.0, *)) {
+    availableBounds = UIEdgeInsetsInsetRect(availableBounds, self.adjustedContentInset);
+  }
+  CGFloat availableWidth = CGRectGetWidth(availableBounds);
+  CGFloat requiredWidth = [self justifiedWidth];
+  return availableWidth >= requiredWidth;
 }
 
-- (void)setBounds:(CGRect)bounds {
-  BOOL shouldScroll =
-      !CGSizeEqualToSize(CGSizeMake(CGRectGetWidth(self.bounds), CGRectGetHeight(self.bounds)),
-                         CGSizeMake(CGRectGetWidth(bounds), CGRectGetHeight(bounds)));
-  [super setBounds:bounds];
-  if (shouldScroll) {
-    [self scrollUntilSelectedItemIsVisibleWithoutAnimation];
+- (void)layoutSubviewsForJustifiedLayout {
+  CGRect availableBounds = self.bounds;
+  if (@available(iOS 11.0, *)) {
+    availableBounds = UIEdgeInsetsInsetRect(availableBounds, self.adjustedContentInset);
   }
-}
 
-- (void)updateConstraints {
-  if (self.containerViewConstraintsActive) {
-    [super updateConstraints];
-    return;
+  BOOL isRTL =
+      [self mdf_effectiveUserInterfaceLayoutDirection] == UIUserInterfaceLayoutDirectionRightToLeft;
+
+  CGFloat itemViewWidth = CGRectGetWidth(availableBounds) / self.itemViews.count;
```

Comment:
If the available width is 1500 and the tab bar has 3 items then the item will being stretched to 500.
Is this what we expected? Or we need to switch to scrollable view in this case..

Reply:
Unless you think it needs to be changed for this version, I'd like to leave the behavior as-is for now.  This logic is simpler and less likely to be wrong.  I think we can improve it in the future, especially since in that case I would expect clients to chose "Clustered Fixed" layout as their preferred layout style.

Reply:
Can this code be reached if the number of items is zero?

Reply:
No longer. Done.